README.md
# README
This repo is used to build the second edition of the book below.

"TrainABA Supervision Curriculum Series Volume 1: BCBA Reference Manual"

## accessibility
It's possible to use SSH keys when added to this repo as a collaborator. Run code below from the local repo clone:

```
git remote set-url origin git@github.com:username/your-repository.git
```
Then try editing a file and do:
```
git add -A
git commit -am "my update msg"
git push
```

### Alternative
You can also clone directly using:
```
git clone git@github.com:username/your-repository.git
```




changelog.md
Changelog

07/21/2019
v 0.0.3.
APA formatting fixed for Bibliography.
\nocite{ref1,ref2} now working in task list subbibliographies. 
Index now includes bibliography components.
Indices divided into Subject Index, Author Index, Title Index.
Added tl.sty, bibAndIndex.sty to style folder.
Cleaned up package list on style.sty.
Added comments to .sty files.
Improved prototype (taskListVanilla.tex) to include dynamic content.
Improved PDF output sample to reflect current formatting.
Added a few lines to bash script (./mk) to compile indices with each build.


notes/vars/taskListIndex.py
# Variables for 5th Edition Task List, BCBA/BCaBA.
# Formatted for python.
#
# 5 refers to the 5th edition.
# 1 refers to Section 1.
# 2 refers to Section 2.
# Letters refer to task list categories, with their corresponding item numbers.

51   = "Section 1: Foundations"
5a   = "A. Philosophical Underpinnings"
5a1  = "A-1 Identify the goals of behavior analysis as a science (i.e., description, prediction, control)."
5a2  = "A-2 Explain the philosophical assumptions underlying the science of behavior analysis (e.g., selectionism, determinism, empiricism, parsimony, pragmatism)."
5a3  = "A-3 Describe and explain behavior from the perspective of radical behaviorism."
5a4  = "A-4 Distinguish among behaviorism, the experimental analysis of behavior, applied behavior analysis, and professional practice guided by the science of behavior analysis."
5a5  = "A-5 Describe and define the dimensions of applied behavior analysis (Baer, Wolf, & Risley, 1968)."

5b   = "B. Concepts and Principles"
5b1  = "B-1 Define and provide examples of behavior, response, and response class."
5b2  = "B-2 Define and provide examples of stimulus and stimulus class."
5b3  = "B-3 Define and provide examples of respondent and operant conditioning."
5b4  = "B-4 Define and provide examples of positive and negative reinforcement contingencies."
5b5  = "B-5 Define and provide examples of schedules of reinforcement."
5b6  = "B-6 Define and provide examples of positive and negative punishment contingencies."
5b7  = "B-7 Define and provide examples of automatic and socially mediated contingencies."
5b8  = "B-8 Define and provide examples of unconditioned, conditioned, and generalized reinforcers and punishers."
5b9  = "B-9 Define and provide examples of operant extinction."
5b10 = "B-10 Define and provide examples of stimulus control."
5b11 = "B-11 Define and provide examples of discrimination, generalization, and maintenance."
5b12 = "B-12 Define and provide examples of motivating operations."
5b13 = "B-13 Define and provide examples of rule-governed and contingency-shaped behavior."
5b14 = "B-14 Define and provide examples of the verbal operants."
5b15 = "B-15 Define and provide examples of derived stimulus relations."

5c   = "C. Measurement, Data Display, and Interpretation"
5c1  = "C-1 Establish operational definitions of behavior."
5c2  = "C-2 Distinguish among direct, indirect, and product measures of behavior."
5c3  = "C-3 Measure occurrence (e.g., frequency, rate, percentage)."
5c4  = "C-4 Measure temporal dimensions of behavior (e.g., duration, latency, interresponse time)."
5c5  = "C-5 Measure form and strength of behavior (e.g., topography, magnitude)."
5c6  = "C-6 Measure trials to criterion."
5c7  = "C-7 Design and implement sampling procedures (i.e., interval recording, time sampling)."
5c8  = "C-8 Evaluate the validity and reliability of measurement procedures."
5c9  = "C-9 Select a measurement system to obtain representative data given the dimensions of behavior and the logistics of observing and recording."
5c10 = "C-10 Graph data to communicate relevant quantitative relations (e.g., equal-interval graphs, bar graphs, cumulative records)."
5c11 = "C-11 Interpret graphed data."

5d   = "D. Experimental Design"
5d1  = "D-1 Distinguish between dependent and independent variables."
5d2  = "D-2 Distinguish between internal and external validity."
5d3  = "D-3 Identify the defining features of single-subject experimental designs (e.g., individuals serve as their own controls, repeated measures, prediction, verification, replication)."
5d4  = "D-4 Describe the advantages of single-subject experimental designs compared to group designs."
5d5  = "D-5 Use single-subject experimental designs (e.g., reversal, multiple baseline, multielement, changing criterion)."
5d6  = "D-6 Describe rationales for conducting comparative, component, and parametric analyses."


52   = "Section 2: Applications"
5e   = "E. Ethics (Professional and Ethical Compliance Code for Behavior Analysts)"
5e1  = "E-1 Responsible conduct of behavior analysts"
5e2  = "E-2 Behavior analysts' responsibility to clients"
5e3  = "E-3 Assessing behavior"
5e4  = "E-4 Behavior analysts and the behavior-change program"
5e5  = "E-5 Behavior analysts as supervisors"
5e6  = "E-6 Behavior analysts' ethical responsibility to the profession of behavior analysis"
5e7  = "E-7 Behavior analysts' ethical responsibility to colleagues"
5e8  = "E-8 Public statements"
5e9  = "E-9 Behavior analysts and research"
5e10 = "E-10 Behavior analysts' ethical responsibility to the BACB"

5f   = "F. Behavior Assessment"
5f1  = "F-1 Review records and available data (e.g., educational, medical, historical) at the outset of the case."
5f2  = "F-2 Determine the need for behavior-analytic services."
5f3  = "F-3 Identify and prioritize socially significant behavior-change goals."
5f4  = "F-4 Conduct assessments of relevant skill strengths and deficits."
5f5  = "F-5 Conduct preference assessments."
5f6  = "F-6 Describe the common functions of problem behavior."
5f7  = "F-7 Conduct a descriptive assessment of problem behavior. "
5f8  = "F-8 Conduct a functional analysis of problem behavior."
5f9  = "F-9 Interpret functional assessment data. "

5g   = "G. Behavior-Change Procedures"
5g1  = "G-1 Use positive and negative reinforcement procedures to strengthen behavior."
5g2  = "G-2 Use interventions based on motivating operations and discriminative stimuli."
5g3  = "G-3 Establish and use conditioned reinforcers."
5g4  = "G-4 Use stimulus and response prompts and fading (e.g., errorless, most-to-least, least-to-most, prompt delay, stimulus fading)."
5g5  = "G-5 Use modeling and imitation training."
5g6  = "G-6 Use instructions and rules."
5g7  = "G-7 Use shaping."
5g8  = "G-8 Use chaining."
5g9  = "G-9 Use discrete-trial, free-operant, and naturalistic teaching arrangements."
5g10 = "G-10 Teach simple and conditional discriminations."
5g11 = "G-11 Use Skinner's analysis to teach verbal behavior."
5g12 = "G-12 Use equivalence-based instruction."
5g13 = "G-13 Use the high-probability instructional sequence."
5g14 = "G-14 Use reinforcement procedures to weaken behavior (e.g., DRA, FCT, DRO, DRL, NCR)."
5g15 = "G-15 Use extinction."
5g16 = "G-16 Use positive and negative punishment (e.g., time-out, response cost, overcorrection)."
5g17 = "G-17 Use token economies."
5g18 = "G-18 Use group contingencies."
5g19 = "G-19 Use contingency contracting."
5g20 = "G-20 Use self-management strategies."
5g21 = "G-21 Use procedures to promote stimulus and response generalization."
5g22 = "G-22 Use procedures to promote maintenance."

5h   = "H. Selecting and Implementing Interventions"
5h1  = "H-1 State intervention goals in observable and measurable terms. "
5h2  = "H-2 Identify potential interventions based on assessment results and the best available scientific evidence."
5h3  = "H-3 Recommend intervention goals and strategies based on such factors as client preferences, supporting environments, risks, constraints, and social validity."
5h4  = "H-4 When a target behavior is to be decreased, select an acceptable alternative behavior to be established or increased."
5h5  = "H-5 Plan for possible unwanted effects when using reinforcement, extinction, and punishment procedures."
5h6  = "H-6 Monitor client progress and treatment integrity."
5h7  = "H-7 Make data-based decisions about the effectiveness of the intervention and the need for treatment revision."
5h8  = "H-8 Make data-based decisions about the need for ongoing services."
5h9  = "H-9 Collaborate with others who support and/or provide services to clients."

5i   = "I. Personnel Supervision and Management"
5i1  = "I-1 State the reasons for using behavior-analytic supervision and the potential risks of ineffective supervision (e.g., poor client outcomes, poor supervisee performance)."
5i2  = "I-2 Establish clear performance expectations for the supervisor and supervisee."
5i3  = "I-3 Select supervision goals based on an assessment of the supervisee's skills."
5i4  = "I-4 Train personnel to competently perform assessment and intervention procedures."
5i5  = "I-5 Use performance monitoring, feedback, and reinforcement systems."
5i6  = "I-6 Use a functional assessment approach (e.g., performance diagnostics) to identify variables affecting personnel performance."
5i7  = "I-7 Use function-based strategies to improve personnel performance."
5i8  = "I-8 Evaluate the effects of supervision (e.g., on client outcomes, on supervisee repertoires)."


notes/vars/tlConcepts.py
# Variables for 5th Edition Task List, BCBA/BCaBA.
# Formatted for python.
#
# 5 refers to the 5th edition.
# 1 refers to Section 1.
# 2 refers to Section 2.
# Letters refer to task list categories, with their corresponding item numbers.

51   = "Section 1: Foundations"
5a   = "A. Philosophical Underpinnings"
5a1  = "A-1 Identify the goals of behavior analysis as a science (i.e., description, prediction, control)."
5a2  = "A-2 Explain the philosophical assumptions underlying the science of behavior analysis (e.g., selectionism, determinism, empiricism, parsimony, pragmatism)."
5a3  = "A-3 Describe and explain behavior from the perspective of radical behaviorism."
5a4  = "A-4 Distinguish among behaviorism, the experimental analysis of behavior, applied behavior analysis, and professional practice guided by the science of behavior analysis."
5a5  = "A-5 Describe and define the dimensions of applied behavior analysis (Baer, Wolf, & Risley, 1968)."
5a5con = ""
5a5ex  = ""
5a5rel = ""
5a5lit = ""

5b   = "B. Concepts and Principles"
5b1  = "B-1 Define and provide examples of behavior, response, and response class."
5b2  = "B-2 Define and provide examples of stimulus and stimulus class."
5b3  = "B-3 Define and provide examples of respondent and operant conditioning."
5b4  = "B-4 Define and provide examples of positive and negative reinforcement contingencies."
5b5  = "B-5 Define and provide examples of schedules of reinforcement."
5b6  = "B-6 Define and provide examples of positive and negative punishment contingencies."
5b7  = "B-7 Define and provide examples of automatic and socially mediated contingencies."
5b8  = "B-8 Define and provide examples of unconditioned, conditioned, and generalized reinforcers and punishers."
5b9  = "B-9 Define and provide examples of operant extinction."
5b10 = "B-10 Define and provide examples of stimulus control."
5b11 = "B-11 Define and provide examples of discrimination, generalization, and maintenance."
5b12 = "B-12 Define and provide examples of motivating operations."
5b13 = "B-13 Define and provide examples of rule-governed and contingency-shaped behavior."
5b14 = "B-14 Define and provide examples of the verbal operants."
5b15 = "B-15 Define and provide examples of derived stimulus relations."

5c   = "C. Measurement, Data Display, and Interpretation"
5c1  = "C-1 Establish operational definitions of behavior."
5c2  = "C-2 Distinguish among direct, indirect, and product measures of behavior."
5c3  = "C-3 Measure occurrence (e.g., frequency, rate, percentage)."
5c4  = "C-4 Measure temporal dimensions of behavior (e.g., duration, latency, interresponse time)."
5c5  = "C-5 Measure form and strength of behavior (e.g., topography, magnitude)."
5c6  = "C-6 Measure trials to criterion."
5c7  = "C-7 Design and implement sampling procedures (i.e., interval recording, time sampling)."
5c8  = "C-8 Evaluate the validity and reliability of measurement procedures."
5c9  = "C-9 Select a measurement system to obtain representative data given the dimensions of behavior and the logistics of observing and recording."
5c10 = "C-10 Graph data to communicate relevant quantitative relations (e.g., equal-interval graphs, bar graphs, cumulative records)."
5c11 = "C-11 Interpret graphed data."

5d   = "D. Experimental Design"
5d1  = "D-1 Distinguish between dependent and independent variables."
5d2  = "D-2 Distinguish between internal and external validity."
5d3  = "D-3 Identify the defining features of single-subject experimental designs (e.g., individuals serve as their own controls, repeated measures, prediction, verification, replication)."
5d4  = "D-4 Describe the advantages of single-subject experimental designs compared to group designs."
5d5  = "D-5 Use single-subject experimental designs (e.g., reversal, multiple baseline, multielement, changing criterion)."
5d6  = "D-6 Describe rationales for conducting comparative, component, and parametric analyses."


52   = "Section 2: Applications"
5e   = "E. Ethics (Professional and Ethical Compliance Code for Behavior Analysts"
5e1  = "E-1 Responsible conduct of behavior analysts"
5e2  = "E-2 Behavior analysts' responsibility to clients"
5e3  = "E-3 Assessing behavior"
5e4  = "E-4 Behavior analysts and the behavior-change program"
5e5  = "E-5 Behavior analysts as supervisors"
5e6  = "E-6 Behavior analysts' ethical responsibility to the profession of behavior analysis"
5e7  = "E-7 Behavior analysts' ethical responsibility to colleagues"
5e8  = "E-8 Public statements"
5e9  = "E-9 Behavior analysts and research"
5e10 = "E-10 Behavior analysts' ethical responsibility to the BACB"

5f   = "F. Behavior Assessment"
5f1  = "F-1 Review records and available data (e.g., educational, medical, historical) at the outset of the case."
5f2  = "F-2 Determine the need for behavior-analytic services."
5f3  = "F-3 Identify and prioritize socially significant behavior-change goals."
5f4  = "F-4 Conduct assessments of relevant skill strengths and deficits."
5f5  = "F-5 Conduct preference assessments."
5f6  = "F-6 Describe the common functions of problem behavior."
5f7  = "F-7 Conduct a descriptive assessment of problem behavior. "
5f8  = "F-8 Conduct a functional analysis of problem behavior."
5f9  = "F-9 Interpret functional assessment data. "

5g   = "G. Behavior-Change Procedures"
5g1  = "G-1 Use positive and negative reinforcement procedures to strengthen behavior."
5g2  = "G-2 Use interventions based on motivating operations and discriminative stimuli."
5g3  = "G-3 Establish and use conditioned reinforcers."
5g4  = "G-4 Use stimulus and response prompts and fading (e.g., errorless, most-to-least, least-to-most, prompt delay, stimulus fading)."
5g5  = "G-5 Use modeling and imitation training."
5g6  = "G-6 Use instructions and rules."
5g7  = "G-7 Use shaping."
5g8  = "G-8 Use chaining."
5g9  = "G-9 Use discrete-trial, free-operant, and naturalistic teaching arrangements."
5g10 = "G-10 Teach simple and conditional discriminations."
5g11 = "G-11 Use Skinner's analysis to teach verbal behavior."
5g12 = "G-12 Use equivalence-based instruction."
5g13 = "G-13 Use the high-probability instructional sequence."
5g14 = "G-14 Use reinforcement procedures to weaken behavior (e.g., DRA, FCT, DRO, DRL, NCR)."
5g15 = "G-15 Use extinction."
5g16 = "G-16 Use positive and negative punishment (e.g., time-out, response cost, overcorrection)."
5g17 = "G-17 Use token economies."
5g18 = "G-18 Use group contingencies."
5g19 = "G-19 Use contingency contracting."
5g20 = "G-20 Use self-management strategies."
5g21 = "G-21 Use procedures to promote stimulus and response generalization."
5g22 = "G-22 Use procedures to promote maintenance."

5h   = "H. Selecting and Implementing Interventions"
5h1  = "H-1 State intervention goals in observable and measurable terms. "
5h2  = "H-2 Identify potential interventions based on assessment results and the best available scientific evidence."
5h3  = "H-3 Recommend intervention goals and strategies based on such factors as client preferences, supporting environments, risks, constraints, and social validity."
5h4  = "H-4 When a target behavior is to be decreased, select an acceptable alternative behavior to be established or increased."
5h5  = "H-5 Plan for possible unwanted effects when using reinforcement, extinction, and punishment procedures."
5h6  = "H-6 Monitor client progress and treatment integrity."
5h7  = "H-7 Make data-based decisions about the effectiveness of the intervention and the need for treatment revision."
5h8  = "H-8 Make data-based decisions about the need for ongoing services."
5h9  = "H-9 Collaborate with others who support and/or provide services to clients."

5i   = "I. Personnel Supervision and Management"
5i1  = "I-1 State the reasons for using behavior-analytic supervision and the potential risks of ineffective supervision (e.g., poor client outcomes, poor supervisee performance)."
5i2  = "I-2 Establish clear performance expectations for the supervisor and supervisee."
5i3  = "I-3 Select supervision goals based on an assessment of the supervisee's skills."
5i4  = "I-4 Train personnel to competently perform assessment and intervention procedures."
5i5  = "I-5 Use performance monitoring, feedback, and reinforcement systems."
5i6  = "I-6 Use a functional assessment approach (e.g., performance diagnostics) to identify variables affecting personnel performance."
5i7  = "I-7 Use function-based strategies to improve personnel performance."
5i8  = "I-8 Evaluate the effects of supervision (e.g., on client outcomes, on supervisee repertoires)."
